### PR TITLE
Refactor commands and arg parsing to use clap

### PR DESCRIPTION
**Pull request by reportingissue**
Will make it easier in the future to add/change commands

### Improvements
- Auto generated help/version
- Automatic argument validation and parsing

### Losses
- No shorthand for commands ("connection" -> "c")

### Questions
- Do we still need `oneshot` and `test`?

### CODE REVIEW

**Comment by p1gp1g** [comment]
Thanks, it looks good.

> ### Losses
> 
> * No shorthand for commands ("connection" -> "c")

That's still possible with the following macro for the cli struct : 

```
#[command(infer_subcommands = true)]
```

> ### Questions
> 
> * Do we still need `oneshot` and `test`?

Test is necessary to help some users to debug. oneshot can be removed.

File: src/cli/connection.rs [hunk 1]
```diff
@@ -2,91 +2,59 @@ use crate::{
...
-    );
-}
+        /// Device number
+        #[arg(value_parser = clap::value_parser!(u32).range(1..))]
+        device_id: u32,
+
+        /// Device token
+        password: String,
+
+        /// Unified push endpoint
```
Comment (p1gp1g) [inline src/cli/connection.rs:21]:
```suggestion
        /// UnifiedPush endpoint
```

File: src/cli/test.rs [hunk 2]
```diff
@@ -1,39 +1,26 @@
...
-{} test uuid aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa
-",
-        env::args().nth(0).unwrap(),
-        env::args().nth(0).unwrap(),
-    );
+#[derive(Subcommand)]
+pub enum TestCommand {
+    /// Test allowed unified push endpoint
+    Endpoint {
+        /// Unified push endpoint
```
Comment (p1gp1g) [inline src/cli/test.rs:9]:
```suggestion
        /// UnifiedPush endpoint
```

File: src/cli/test.rs [hunk 3]
```diff
@@ -1,39 +1,26 @@
...
-Usage:
-{} test endpoint https://push.server.ltd/id
-{} test uuid aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa
-",
-        env::args().nth(0).unwrap(),
-        env::args().nth(0).unwrap(),
-    );
+#[derive(Subcommand)]
+pub enum TestCommand {
+    /// Test allowed unified push endpoint
```
Comment (p1gp1g) [inline src/cli/test.rs:7]:
```suggestion
    /// Test allowed UnifiedPush endpoint
```

File: src/cli.rs [hunk 4]
```diff
@@ -1,37 +1,64 @@
...
 mod oneshot;
 mod server;
 mod test;
 
-fn usage() {
-    println!(
-        "
-Usage: {0} [command] [args, ...]
+#[derive(Parser)]
+#[command(author, version, about, long_about = None)]
```
Comment (p1gp1g) [inline src/cli.rs:15]:
```suggestion
#[command(author, version, about, long_about = None)]
#[command(infer_subcommands = true)]
```

**Comment by reportingissue** [comment]
Ty for review and `infer_subcommands`, `oneshot` has been removed

**Comment by p1gp1g** [comment]
Thanks :)

**Pull request merged**